File: Algorithms/Python/0003/0003.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if s == "":
            return 0
        map, res, i = {}, 0, -1
        for j in range(len(s)):
            if s[j] in map and map[s[j]] > 0:
                while True:
                    i = i + 1
                    if s[i] == s[j]:
                        break
                    else:
                        map[s[i]] = 0
            else:
                map[s[j]] = 1
            res = max(res, j - i)
        return res

File: Algorithms/C++/0003/0003.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s == "") {
            return 0;
        }
        map<char, int> m;
        int res = 0, i = -1;
        for (int j = 0; j < s.size(); j++) {
            if (m[s[j]] > 0) {
                while(s[++i] != s[j]) {
                    m[s[i]]--;
                }
            } else {
                m[s[j]]++;
            }
            res = max(res, j - i);
        }
        return res;
    }
};
